lib/CL/clEnqueueNDRangeKernel.c

```diff
@@ -230,50 +228,25 @@ POname(clEnqueueNDRangeKernel)(cl_command_queue command_queue,
   POCL_RETURN_ERROR_COND((event_wait_list != NULL && num_events_in_wait_list == 0),
     CL_INVALID_EVENT_WAIT_LIST);
 
+  cache_lock = pocl_cache_acquire_writer_lock(kernel->program,
+                                              command_queue->device);
+  assert(cache_lock);
```

Comment:
Is this the situation that can happen after someone hits ctrl-c during a kernel build and runs the program again? I think a more descriptive error message (how to recover from this) should be in place.